src/Configuration/RectorConfigBuilder.php

```diff
@@ -962,6 +972,41 @@ public function withTypeCoverageLevel(int $level): self
         return $this;
     }
 
+    /**
+     * @experimental Since 1.2.5 Raise your PHP level from, one level at a time
+     */
+    public function withPhpLevel(int $level): self
+    {
+        Assert::natural($level);
+
+        $this->isWithPhpLevelUsed = true;
+
+        $phpVersion = ComposerJsonPhpVersionResolver::resolveFromCwdOrFail();
```

Comment:
I think this should allow read from `withPhpVersion()` defined from Rector config so it can be tested on getrector.com, which PhpVersionProvider seems can be used, or add fallback on `ComposerJsonPhpVersionResolver` to use predefined from withPhpversion() config when exists

Reply:
I created experiment PR at https://github.com/rectorphp/rector-src/pull/6384